src/main/java/de/siegmar/logbackgelf/GelfEncoder.java

```diff
@@ -439,33 +439,12 @@ protected String normalizeShortMessage(final String shortMessage) {
     }
 
     private String sanitizeShortMessage(final String sanitizedShortMessage) {
-        if (sanitizedShortMessage.isEmpty()) {
-            return sanitizedShortMessage;
+        final String stripped = sanitizedShortMessage.strip();
+        if (getMaxShortMessageLength() != 0 && stripped.length() > getMaxShortMessageLength()) {
+            return stripped.substring(0, getMaxShortMessageLength());
```

Comment:
A message `a          b` with a max length of 5 would be returned as `a    `. This is why I'd first only strip leading whitespaces, then do an optional substring and then the stripping of trailing whitespaces.